erpnext/stock/get_item_details.py

```diff
@@ -309,8 +309,7 @@ def get_basic_details(args, item, overwrite_warehouse=True):
 	if item.stock_uom == args.uom:
 		out.conversion_factor = 1.0
 	else:
-		out.conversion_factor = args.conversion_factor or \
-			get_conversion_factor(item.name, args.uom).get("conversion_factor")
+		out.conversion_factor = get_conversion_factor(item.name, args.uom).get("conversion_factor")
```

Comment:
The fix is not correct, because this field is editable and user can add manually conversion factor in case conversion factor is not set against the selected UOM. With above changes system will set value again to zero even if user has manually set the conversion factor.

The correct fix can be, on selection of another item at JS side set the conversion factor as zero so that system will recalculate the conversion factor.